src/traits/impls.rs

```diff
@@ -42,6 +51,27 @@ impl DoughnutApi for () {
     }
 }
 
+#[cfg(feature = "std")]
+impl Signing for DoughnutV0 {
+    fn sign_ed25519(&mut self, secret_key: &[u8]) -> Result<H512, SigningError> {
+        sign_ed25519(&self.issuer(), secret_key, &self.payload())
```

Comment:
Will need to set the signature version in these functions also
`self.signature_version = SignatureVersion::Ed25519 as u8;`
Could add some assertion to check signature version in tests also